psp/dataset.py

```diff
@@ -193,23 +207,22 @@ def split_train_test(
         valid_split: Ratio of Pv sites from the train set to use as valid set. Note that the
             time range is the same for train and valid.
     """
-    # For simplicity, we only support setting all the dates or None of them, in which we simply
-    # split the data 50/50 between train and test.
-    num_none = sum([x is None for x in [train_start, train_end, test_start, test_end]])
-
-    if num_none not in [0, 4]:
-        raise NotImplementedError("you need to specify all dates or no dates.")
+    # Use some simple rules to choose missing dates.
+    if train_start is None:
+        train_start = _ceiling_date(pv_data_source.min_ts())
 
-    if num_none == 4:
-        min_ts = pv_data_source.min_ts()
-        max_ts = pv_data_source.max_ts()
+    if test_end is None:
+        test_end = _floor_date(pv_data_source.max_ts())
 
-        train_start = min_ts
-        test_end = max_ts
+    if train_end is None and test_start is not None:
+        train_end = test_start - timedelta(days=1)
+    elif test_start is None and train_end is not None:
+        test_start = train_end + timedelta(days=1)
+    elif train_end is None and test_start is None:
```

Comment:
Looks good

Reply:
wouldn't you want to put the

    elif test_start is None and train_end is not None:
        test_start = train_end + timedelta(days=1)

before:

    if train_end is None and test_start is not None:
        train_end = test_start - timedelta(days=1)

Reply:
as you have not yet defined what test_start should be before using it?

Reply:
It doesn't change anything because those 2 are mutually exclusive... I think :D I agree they are a bit confusing